packages/jellyfish-api-core/__tests__/category/account/removePoolLiquidity.test.ts

```diff
@@ -0,0 +1,112 @@
+import { ContainerAdapterClient } from '../../container_adapter_client'
+
+describe('Poolpair', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+  let poolLiquidityAddress: string
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+    await container.waitForWalletBalanceGTE(200)
+
+    await createToken('DDAI')
+    await mintTokens('DDAI')
+    await createPoolPair('DDAI')
+    await addPoolLiquidity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  async function createToken (symbol: string): Promise<void> {
+    const address = await container.call('getnewaddress')
+    const metadata = {
+      symbol,
+      name: symbol,
+      isDAT: true,
+      mintable: true,
+      tradeable: true,
+      collateralAddress: address
+    }
+    await container.call('createtoken', [metadata])
+    await container.generate(1)
+  }
+
+  async function createPoolPair (tokenB: string, metadata?: any): Promise<void> {
+    const address = await container.call('getnewaddress')
+    const defaultMetadata = {
+      tokenA: 'DFI',
+      tokenB: 'DDAI',
+      commission: 0,
+      status: true,
+      ownerAddress: address
+    }
+    await client.poolpair.createPoolPair({ ...defaultMetadata, ...metadata })
+    await container.generate(1)
+  }
+
+  async function mintTokens (symbol: string): Promise<void> {
+    const address = await container.call('getnewaddress')
+    const payload: { [key: string]: string } = {}
+    payload[address] = '100@0'
+    await container.call('utxostoaccount', [payload])
+    await container.call('minttokens', [`2000@${symbol}`])
+    await container.generate(1)
+  }
+
+  async function addPoolLiquidity (): Promise<void> {
+    const shareAddress = await container.call('getnewaddress')
+    const data = await client.poolpair.addPoolLiquidity({
+      '*': ['10@DFI', '200@DDAI']
+    }, shareAddress)
+    await container.generate(1)
+
+    expect(typeof data).toStrictEqual('string')
+  }
+
+  it('should removePoolLiquidity', async () => {
+    const poolPairBefore = await container.call('listpoolpairs')
+    console.log('poolpairs: ', poolPairBefore)
+
+    const totalLiquidityBefore = poolPairBefore['2'].totalLiquidity
+
+    try {
+      await container.call('removepoolliquidity', [poolLiquidityAddress, '13@DFI-DDAI'])
+      await container.generate(1)
+    } catch (err) {
+      console.log('err: ', err)
+    }
+
+    const poolPairAfter = await container.call('listpoolpairs')
+    console.log('poolpairs: ', poolPairAfter)
+    const totalLiquidityAfter = poolPairAfter['2'].totalLiquidity
+
+    // expect(poolPairAfter['2'].totalLiquidity - poolPairBefore['2'].totalLiquidity.toStrictEqual(13))
+    expect(totalLiquidityBefore - totalLiquidityAfter).toStrictEqual(0) // 13
```

Comment:
hmm.. i feel not right here... should not be zero
eg:
before remove - totalLiquidity 100
remove 29 liquidity
after remove - totalLiquidity 71

Reply:
@canonbrother Do you mind elaborating further on this please? I understand the logic behind it, so we would have to change `0` to something like `BigNumber`?

Reply:
@canonbrother what does this error mean? 
`TypeError: Cannot read property 'totalLiquidity' of undefined`